linera-summary/src/github.rs

```diff
@@ -164,20 +166,66 @@ impl Github {
         &self.context
     }
 
-    pub async fn comment_on_pr(&self, body: String) -> Result<()> {
+    pub async fn upsert_pr_comment(&self, body: String) -> Result<()> {
+        let issue_handler = self.octocrab.issues(
+            self.context.repository.owner.clone(),
+            self.context.repository.name.clone(),
+        );
+        let comments = issue_handler
+            .list_comments(self.context.pr_number)
+            .send()
+            .await?;
+        let existing_comments = comments
+            .items
+            .into_iter()
+            .filter(|comment| {
+                comment.user.login == "github-actions[bot]"
+                    && comment
+                        .body
+                        .as_ref()
+                        .is_some_and(|body| body.starts_with(PR_COMMENT_HEADER))
+            })
+            .collect::<Vec<_>>();
```

Comment:
Optional: instead of using `filter`, `collect` and `first`, this could just be `find`. It then returns `Option`, which would then require `if let Some(existing_comment_id)` further below.

```suggestion
            .find(|comment| {
                comment.user.login == "github-actions[bot]"
                    && comment
                        .body
                        .as_ref()
                        .is_some_and(|body| body.starts_with(PR_COMMENT_HEADER))
            });
```

Reply:
Ah, good point!